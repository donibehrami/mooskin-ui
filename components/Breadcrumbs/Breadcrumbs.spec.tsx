import * as React from 'react';
import BreadcrumbsGroup, {Breadcrumb} from './Breadcrumbs';

import { render, shallow} from 'enzyme';

describe('Breadcrumbs', () => {

    test('renders correctly', () => {
        const tree = shallow(
            <BreadcrumbsGroup>
                <Breadcrumb id="1" title="Breadcrumb 1" active>
                    Content 1
                </Breadcrumb>
                <Breadcrumb id="2" title="Breadcrumb 2">
                    Content 2
                </Breadcrumb>
                <Breadcrumb id="3" title="Breadcrumb 3">
                    Content 3
                </Breadcrumb>
            </BreadcrumbsGroup>
        );
        expect(tree).toMatchSnapshot();
    });

    test('renders correctly with only the active one being visible', () => {
        const tree = render(
            <BreadcrumbsGroup>
                <Breadcrumb id="1" title="Breadcrumb 1" active>
                    Content 1
                </Breadcrumb>
                <Breadcrumb id="2" title="Breadcrumb 2">
                    Content 2
                </Breadcrumb>
                <Breadcrumb id="3" title="Breadcrumb 3">
                    Content 3
                </Breadcrumb>
            </BreadcrumbsGroup>
        );

        expect(tree.find('.activeBreadcrumb').length).toBe(1);
    });

    test('renders the body correctly with each content', () => {
        const tree = render(
            <BreadcrumbsGroup>
                <Breadcrumb id="1" title="Breadcrumb 1" active>
                    Content 1
                </Breadcrumb>
                <Breadcrumb id="2" title="Breadcrumb 2">
                    Content 2
                </Breadcrumb>
                <Breadcrumb id="3" title="Breadcrumb 3">
                    Content 3
                </Breadcrumb>
            </BreadcrumbsGroup>
        );

        expect(tree.find('.breadcrumbBody').length).toBe(3);
        expect(tree.find('.breadcrumbBody').first().text()).toContain('Content 1');
        expect(tree.find('.breadcrumbBody').eq(1).text()).toContain('Content 2');
        expect(tree.find('.breadcrumbBody').last().text()).toContain('Content 3');
    });

    test('onClick is called correctly when one of the breadcrumbs is clicked', () => {

        const fn = jest.fn();

        const tree = shallow(
            <BreadcrumbsGroup>
                <Breadcrumb id="1" title="Breadcrumb 1" active>
                    Content 1
                </Breadcrumb>
                <Breadcrumb id="2" title="Breadcrumb 2" onClick={fn}>
                    Content 2
                </Breadcrumb>
                <Breadcrumb id="3" title="Breadcrumb 3">
                    Content 3
                </Breadcrumb>
            </BreadcrumbsGroup>
        );

        tree.find(Breadcrumb)
            .findWhere((bc) => bc.prop('id') === '2')
            .dive()
            .find('#2')
            .simulate('click');

        expect(fn).toHaveBeenCalledWith('2');
    });
});
