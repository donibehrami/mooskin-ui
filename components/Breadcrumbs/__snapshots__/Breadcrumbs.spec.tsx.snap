// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs renders correctly 1`] = `
<div
  className="breadcrumbsGroup "
>
  <div
    className="breadcrumbsContainer"
  >
    <Breadcrumb
      active={true}
      id="1"
      title="Breadcrumb 1"
    >
      Content 1
    </Breadcrumb>
    <Breadcrumb
      id="2"
      title="Breadcrumb 2"
    >
      Content 2
    </Breadcrumb>
    <Breadcrumb
      id="3"
      title="Breadcrumb 3"
    >
      Content 3
    </Breadcrumb>
  </div>
  <div>
    <BreadcrumbBody
      active={true}
      content="Content 1"
    />
    <BreadcrumbBody
      content="Content 2"
    />
    <BreadcrumbBody
      content="Content 3"
    />
  </div>
</div>
`;
